scripts/zones/AlTaieu/mobs/Absolute_Virtue.lua

```diff
@@ -4,20 +4,274 @@
 -----------------------------------
 local ID = zones[xi.zone.ALTAIEU]
 -----------------------------------
+
+--[[
+    misc sources:
+    http://rukenshin.livejournal.com/15173.html
+
+    Post-nerf HP: 11k heal at 95% HP. 11000 / (1 - .95) = 220,000
+    http://home.eyesonff.com/showthread.php/80870-Absolute-Virtue-Beaten
+
+    Stats: +100 MDB, +132 MAB
+    https://www.bluegartr.com/threads/81211-Absolute-Virtue-Information-(OP-Updated)
+
+    - Near instant to instant fast cast
+    - Enhanced Movement Speed (~150%)
+    - Immune to all enfeebles except Shadowbind, Elemental DoTs, Dia, and Bio [?? - any other exceptions?]
+    - Quickly gains resistance to Souleater damage (September '08 patch)
+    - Resists Modus Veritas (October '09 patch)
+
+    In addition to the magic defense bonus +100 outlined earlier, AV has a general all damage types reduced that gradually
+    increases as its current HP decreases. This applies to all forms of damage - magic, physical, and even non-elemental
+    such as Shield Bash. This damage reduction seems to increase proportionally to its remaining HP%
+    (for instance, at 50% HP, it has roughly -50% damage taken on top of MDB). It is unknown how this modifier works at extremely
+    low HP, since if this reduction remained linear, at 1% HP, it would have 99% damage reduction (virtually unkillable).
+]]
+
+-- instead of storing a bunch of numeric vars we will store a global object
+xi = xi or {}
+xi.av = xi.av or {}
+
+local combos =
+{
+    [xi.jsa.CHAINSPELL] = { xi.jsa.CHAINSPELL, xi.jsa.MANAFONT, { xi.jsa.CHAINSPELL, xi.jsa.SOUL_VOICE } },
+    [xi.jsa.MIGHTY_STRIKES] = { xi.jsa.MIGHTY_STRIKES, xi.jsa.HUNDRED_FISTS },
+    [xi.jsa.MEIKYO_SHISUI] = { xi.jsa.MEIKYO_SHISUI, xi.jsa.EES_AERN, xi.jsa.EES_AERN, xi.jsa.EES_AERN },
+    [xi.jsa.INVINCIBLE] = { xi.jsa.INVINCIBLE, xi.jsa.BENEDICTION, xi.jsa.MIJIN_GAKURE },
+    [xi.jsa.CALL_WYVERN] = { xi.jsa.CALL_WYVERN, xi.jsa.FAMILIAR, xi.jsa.ASTRAL_FLOW },
+}
+
+local debugAV = false -- Set to true to get debug prints about AV's behaviour
+local debug   = utils.getDebugPlayerPrinter(debugAV)
+
 local entity = {}
 
+entity.handleDamageResists = function(mob)
+    local next = mob:getLocalVar('dmgThreshold')
+    local current = mob:getHPP()
+    if current <= next then
+        mob:setLocalVar('dmgThreshold', next - 10)
+        local dmg = (100 - current) * -1
+        mob:setMod(xi.mod.UDMGPHYS, dmg)
+        mob:setMod(xi.mod.UDMGRANGE, dmg)
+        mob:setMod(xi.mod.UDMGMAGIC, dmg * 10)
+        mob:setMod(xi.mod.UDMGBREATH, dmg)
+    end
+end
+
+local playerAbilityToMobSP =
+{
+    [16] = 688, -- MIGHTY_STRIKES
+    [17] = 690, -- HUNDRED_FISTS
+    [18] = 689, -- BENEDICTION
+    [19] = 691, -- MANAFONT
+    [20] = 692, -- CHAINSPELL
+    [21] = 693, -- PERFECT_DODGE
+    [22] = 694, -- INVINCIBLE
+    [23] = 695, -- BLOOD_WEAPON
+    [24] = 740, -- FAMILIAR
+    [25] = 696, -- SOUL_VOICE
+    [26] = 1389, -- EES_AERN
```

Comment:
super nitpick: this one is out of alignment with the rest

Reply:
Yeah, I hate it too, will fix